src/modules/navigator/navigator_main.cpp

```diff
@@ -310,17 +310,35 @@ void Navigator::run()
 						rep->current.cruising_speed = cmd.param1;
 					}
 
-					rep->current.cruising_throttle = get_cruising_throttle();
-					rep->current.acceptance_radius = get_acceptance_radius();
-
-					// Go on and check which changes had been requested
-					if (PX4_ISFINITE(cmd.param4)) {
-						rep->current.yaw = cmd.param4;
+					// (3) Loiter radius (param3)
+					if (PX4_ISFINITE(cmd.param3) && cmd.param3 > 0.0f) {
+						rep->current.loiter_radius = fabsf(cmd.param3);
 
 					} else {
+						// Use current or default if param3 not valid
+						rep->current.loiter_radius = get_loiter_radius();
+					}
+
+					// (4) Loiter direction / yaw (param4)
+					if (_vstatus.vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING) {
+						// param4 >= 1 => counter-clockwise, else default CW
+						rep->current.loiter_direction_counter_clockwise = (PX4_ISFINITE(cmd.param4) && (cmd.param4 >= 1.0f));
```

Comment:
```suggestion
						rep->current.loiter_direction_counter_clockwise = (PX4_ISFINITE(cmd.param4) && (cmd.param4 >= FLT_EPSILON));
```
That's more robust to detect >0 than comparing against a float 1.